modules/apps/commerce/commerce-product-content-web/src/main/java/com/liferay/commerce/product/content/web/internal/info/item/renderer/CSDiagramEntryInfoItemRenderer.java

```diff
@@ -0,0 +1,114 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.commerce.product.content.web.internal.info.item.renderer;
+
+import com.liferay.commerce.constants.CommerceWebKeys;
+import com.liferay.commerce.context.CommerceContext;
+import com.liferay.commerce.product.content.constants.CPContentWebKeys;
+import com.liferay.commerce.product.model.CPDefinition;
+import com.liferay.commerce.product.permission.CommerceProductViewPermission;
+import com.liferay.commerce.product.service.CPDefinitionLocalService;
+import com.liferay.commerce.shop.by.diagram.model.CSDiagramEntry;
+import com.liferay.commerce.util.CommerceUtil;
+import com.liferay.info.item.renderer.InfoItemRenderer;
+import com.liferay.portal.kernel.language.Language;
+import com.liferay.portal.kernel.theme.ThemeDisplay;
+import com.liferay.portal.kernel.util.WebKeys;
+
+import java.util.Locale;
+
+import javax.servlet.RequestDispatcher;
+import javax.servlet.ServletContext;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Mahmoud Azzam
+ */
+@Component(
+	service = {CSDiagramEntryInfoItemRenderer.class, InfoItemRenderer.class}
+)
+public class CSDiagramEntryInfoItemRenderer
+	implements InfoItemRenderer<CSDiagramEntry> {
+
+	@Override
+	public String getLabel(Locale locale) {
+		return _language.get(locale, "related-diagrams");
+	}
+
+	@Override
+	public void render(
+		CSDiagramEntry csDiagramEntry, HttpServletRequest httpServletRequest,
+		HttpServletResponse httpServletResponse) {
+
+		if (csDiagramEntry == null) {
+			return;
+		}
+
+		try {
+			CPDefinition cpDefinition =
+				_cpDefinitionLocalService.fetchCPDefinitionByCProductId(
+					csDiagramEntry.getCProductId());
+
+			CommerceContext commerceContext =
+				(CommerceContext)httpServletRequest.getAttribute(
+					CommerceWebKeys.COMMERCE_CONTEXT);
+
+			ThemeDisplay themeDisplay =
+				(ThemeDisplay)httpServletRequest.getAttribute(
+					WebKeys.THEME_DISPLAY);
+
+			if ((cpDefinition == null) ||
```

Comment:
In order to optimize the code, let's do the null check before getting the `commerceContext` and the `themeDisplay`